crypto/ct/ct_sct.c

```diff
@@ -317,12 +317,11 @@ int SCT_set_source(SCT *sct, sct_source_t source)
 
 int SCT_LIST_set_source(const STACK_OF(SCT) *scts, sct_source_t source)
 {
-    int i, ret = 1;
+    int i, ret = 0;
 
     for (i = 0; i < sk_SCT_num(scts); ++i) {
-        ret = SCT_set_source(sk_SCT_value(scts, i), source);
-        if (ret != 1)
-            break;
+        if (SCT_set_source(sk_SCT_value(scts, i), source))
```

Comment:
The caller could report that there was a problem with _n_ SCTs, for instance. It could identify all of the problematic SCTs with a single loop. Also, all of the other SCTs would be immediately usable, having had their source set correctly, even if they appeared after a problematic SCT in the list. This generally seems like a better state of affairs than not even trying to set the source of the later SCTs. The error isn't really swallowed, as __owur requires that the return value be checked and so the caller should identify any case where an error occurred.

Admittedly, this is something of a moot point, given that the only thing that will cause `SCT_LIST_set_source` and `SCT_set_source` to fail at the moment is passing an invalid source. This might change in the future though, with CT v2 on the horizon.

I'm happy with either way of doing it though. Failing early and returning 0 is certainly a little simpler, so maybe it should win on those grounds alone.